src/service/git/git-api-service.ts

```diff
@@ -125,15 +125,18 @@ export class GitAPIService {
   ): Promise<string> {
     try {
       // check whether there is a fork with the same name as repo name
-      const repoName = await this.checkIfRepositoryExists(targetOwner, repo);
+      this.logger.info(`Checking if ${targetOwner}/${repo} is forked to ${sourceOwner}/${repo}`);
+      const repoName = await this.checkIfRepositoryExists(sourceOwner, repo);
```

Comment:
@jstastny-cz @Ginxo TBH I'm not sure if using `checkIfRepositoryExists` with `sourceOwner` instead of `targetOwner` is the correct way to go, after this change I'm getting many failures in the e2e tests locally and seems that in GH they are running forever and never finish.

To run them locally I used:
```
$ npm ci
$ ACT_LOG=true npm run test:e2e
```

Actually taking a look in the existing [single-pr.test.ts](https://github.com/kiegroup/github-action-build-chain/blob/main/test/e2e/single-pr/single-pr.test.ts#L352) e2e test, there is test `PR from owner2/target:branchA to owner1/target-different-name:branchB` which should be testing the case where the fork has a different name from the original repository and the test is passing. Am I missing something?

Reply:
that's it, this functionality is [already covered](https://github.com/kiegroup/github-action-build-chain/blob/52ecca89476d3b42829d20fe72eeb3b39aa4dc5c/src/service/git/git-api-service.ts#L121) and it shouldn't be re-adapted unless we discover specific corner case for it I'm not able to find (can't get trace logs). 
I'm afraid we are missing something on the BC configuration rather than BC bug, but it is just a feeling I would like to confirm/discard.

Reply:
Does it really seem correct to you @Ginxo ? We want to check here in the method if there's a fork with the same name as repo name  ... but why would we check that the repo in target exists as the first thing and if so then return? 

In code we invoke `getForkName(currentTarget.group, originalSource.group!, currentTarget.name)` in checkout-service.ts:L163, where the currentTarget variable in same file L140 is commented as 
>      // map the starting project target branch to the corresponding branch defined in the mapping for the current node
 >      // target branch is guaranteed to exist since base always exist
 
 That really sounds like identity check. And the single-pr.test.ts - I am gonna check, but in this case I have mutliple PRs across 3 repositories, maybe it tests a different thing?

Reply:
It certainly does not do what the comment just above it says `// check whether there is a fork with the same name as repo name`

Reply:
Consider this reproducer :
```
test("getForkNameForDifferentName", async () => {
    const moctokit = new Moctokit();
    moctokit.rest.repos
        .get({ owner: "sourceOwner", repo: "repo" })
        .reply({ status: 200, data: {} });
    moctokit.rest.repos
        .get({ owner: "targetOwner", repo: "repo" })
        .reply({ status: 200, data: { name: "repo", owner: { login: "targetOwner" } } });
    moctokit.rest.repos.listForks({ owner: "targetOwner", repo: "repo" }).reply({
        status: 200,
        data: [{ name: "repo-fork", owner: { login: "sourceOwner" } }],
    });
    await expect(git.getForkName("targetOwner", "sourceOwner", "repo")).resolves.toBe(
        "repo-fork"
    );
})
```

getForkName returns "repo".